src/core/lv_obj_style.h

```diff
@@ -155,6 +155,16 @@ void lv_obj_enable_style_refresh(bool en);
  */
 lv_style_value_t lv_obj_get_style_prop(const struct _lv_obj_t * obj, lv_part_t part, lv_style_prop_t prop);
 
+/**
+ * Check if the specified style property is explicitly set for a given object and part.
+ * @param obj   Pointer to an object.
+ * @param part  A part from which the property should be checked.
+ * @param prop  The style property to check.
+ * @return      LV_STYLE_RES_NOT_FOUND: the property wasn't found.
+ *              LV_STYLE_RES_FOUND: the property was fond.
+ */
+lv_style_res_t lv_obj_style_prop_set_check(const struct _lv_obj_t * obj, lv_part_t part, lv_style_prop_t prop);
```

Comment:
```suggestion
bool lv_obj_has_style_prop(const struct _lv_obj_t * obj, lv_style_selector_t selector, lv_style_prop_t prop);
```

I suggest using this prototype:
- Use the `has` keywork in the function name, similarly to `lv_obj_has_flag`
- Use `lv_style_selector_t` to check a part in a given state. E.g. `LV_PART_INDICATOR | LV_STATE_PRESSED`
- Return `bool` as it clearly answers the `has` question.